firedrake/slate/slac/kernel_builder.py

```diff
@@ -125,20 +126,20 @@ def prepare_tsfc_kernels(temps, tsfc_parameters=None):
     for expr in temps.keys():
         integrals = expr.form.integrals()
         mapper = RemoveRestrictions()
-        integrals = map(partial(map_integrand_dags, mapper), integrals)
+        integrals = list(map(partial(map_integrand_dags, mapper), integrals))
         prefix = "subkernel%d_" % len(kernel_exprs)
 
         # Now we split integrals by type: interior_facet and all other cases
         # First, the interior_facet case:
-        interior_facet_intergrals = filter(lambda x: x.integral_type() == "interior_facet", integrals)
+        interior_facet_intergrals = list(filter(lambda x: x.integral_type() == "interior_facet", integrals))
 
         # Now we reconstruct all interior_facet integrals to be of type: exterior_facet
         # This is because locally over each cell, SLATE views them as being "exterior"
         # with respect to the cell.
         interior_facet_intergrals = [it.reconstruct(integral_type="exterior_facet")
                                      for it in interior_facet_intergrals]
         # Now for the rest:
-        other_integrals = filter(lambda x: x.integral_type() != "interior_facet", integrals)
+        other_integrals = list(filter(lambda x: x.integral_type() != "interior_facet", integrals))
```

Comment:
Here you need to coerce to list because `Form` doesn't handle the `iterable` case properly.